### PR TITLE
Refactor tests to work correctly

### PR DESCRIPTION
**Pull request by July541**
This pr includes:

1. call `resolve` and `reject` explicitly.
2. await Promise to make assert workable.
3. Fix an error that keyword to match turns out has been modified in hls, see comment below.
4. Remove `any` and give them an explicit type.

After this merged, we should see #857 failed.

### CODE REVIEW

File: test/suite/extension.test.ts [hunk 1]
```diff
@@ -1,200 +1,208 @@
...
-    const logContents = getExtensionLogContent();
-    assert.ok(logContents, 'Extension log file does not exist');
     assert.ok(
-      retryOperation(
+      await retryOperation(
         () =>
-          new Promise((resolve, reject) =>
-            logContents.match(/INFO hls:\s+Registering ide configuration/) !== null ? resolve : reject
+          new Promise((resolve, reject) => {
+            return getExtensionLogContent()?.match(/Registering ide configuration/i) !== null
```
Comment (July541) [inline test/suite/extension.test.ts:168]: Our keyword to match turns out has been modified in hls...

**Review by fendor — APPROVED** [review]
Awesome, thank you!

**Comment by July541** [comment]
Please don't merge, something is still working on.

**Comment by fendor** [comment]
@July541 Sorry, I misread :( But it doesnt matter, we don't deploy right away, any way :/

**Comment by July541** [comment]
Never mind, I'll fix it in another pr.

**Pull request merged**